### ISSUE TITLE
Switch to new ILSpy DLL's

### ISSUE THREAD

**Issue by jogibear9988** [dnSpyRevived/dnSpy#2]
This issue was also open on the original dnSpy repo.

@dgrunwald the maintainer of ILSpy also wrote something about what needs to be done.

With the old Version used in the project, many code could not be decompiled.

But I think this would be a hard task.

**Comment by IntelOrca** [comment]
I haven't looked at what changes dnSpy made to the fork, so I am not sure. Do you know why you think it would be hard? Are there incompatible API changes?

**Comment by jogibear9988** [comment]
Don't know.
As I said, dgrundwald wrote to the dnSpy issue.

**Comment by dgrunwald** [comment]
dnSpy is using a significantly patched version of ICSharpCode.Decompiler version 2.x. The original decompiler used Mono.Cecil, the patched version used dnlib instead.

ICSharpCode.Decompiler version 3 was an almost-complete rewrite (https://github.com/icsharpcode/ILSpy/wiki/newdecompiler), which makes an upgrade tricky.

On top of that, ICSharpCode.Decompiler version 4 switched from Mono.Cecil to System.Reflection.Metadata (https://github.com/icsharpcode/ILSpy/wiki/srm). SRM has a significantly different API style from Mono.Cecil/dnlib (more direct access to the metadata tables; no mutable object model). This makes it basically impossible to "just replace SRM with dnlib" the way dnspy did before.

My recommendation is to use ICSharpCode.Decompiler as-is and just accept that dnspy will need to load assemblies twice (with dnlib + SRM).

**Comment by rollsch** [comment]
Has anyone looked into doing this?

**Comment by dgrunwald** [comment]
Yes, on a different dnSpy-reviving repository: https://github.com/dnSpyEx/dnSpy/issues/5#issuecomment-1019545903
